org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/imports/TypeChooser.java

```diff
@@ -62,7 +62,7 @@ public JvmDeclaredType choose(final List<JvmDeclaredType> candidateTypes, Iterab
 		IStructuredContentProvider contentProvider = new ContentProvider();
 		Dialog dialog = new Dialog(shell, new LabelProvider(labelProvider), contentProvider);
 		dialog.setInput(candidateTypes);
-		dialog.setInitialSelections(new JvmDeclaredType[] { candidateTypes.get(0) });
+		dialog.setInitialSelections((Object[])new JvmDeclaredType[] { candidateTypes.get(0) });
```

Comment:
@szarnekow is this correct to indicate we call the in later eclipse versions variadic param in a non variadic way?

Reply:
Actually I'd think passing an array of JvmDeclaredType did already indicate this and would be source compatible to a varargs-change.

Reply:
well jdt complains.
see your workspace ;)

Reply:
e.g. `Arrays.asList(new String[] { "" });` this is a `List<String>` and not a `List<String[]>`

Reply:
My example was broken since Arrays.asList uses a type parameter.